pages/index.js

```diff
@@ -1,30 +1,79 @@
-import React from "react";
+import React, { useState } from 'react';
+import styled from 'styled-components';
 
-import TodoInput from "../components/todo-input";
-import TodoList from "../components/todo-list";
-
-import styled from "styled-components";
+import TodoInput from '../components/todo-input';
+import TodoList from '../components/todo-list';
 
 export default function Home() {
-  return (
-    <Wrapper>
-      <Contents>
-        <TodoInput />
-        <TodoList />
-      </Contents>
-    </Wrapper>
-  );
+	const [date, setDate] = useState();
+	const [todo, setTodo] = useState();
+	const [todos, setTodos] = useState([]);
+	const [id, setID] = useState(0);
+
+	const changeTodo = (e) => {
+		setTodo(e.target.value);
+	};
+
+	const changeDate = (e) => {
+		setDate(e.target.value);
+	};
+
+	const removeTodo = (id) => {
+		const newTodos = [...todos];
+		setTodos(newTodos.filter((info) => info.id !== id));
+	};
+	const submitTodo = (e) => {
+		e.preventDefault();
+		if (todo === '' || date === '') {
+			alert('모든 항목을 입력해주세요!');
+			return false;
+		} else if (date.length != 8) {
+			alert('날짜를 올바른 형식으로 입력해주세요!');
+			return false;
+		} else alert('입력되었습니다!');
+		setTodos(todos.concat({ id: id, date: date, todo: todo }));
+		setTodo('');
+		setDate('');
+		setID(id + 1);
+	};
+
+	return (
+		<Wrapper>
+			<Subject>리액트-투두</Subject>
+			<Contents>
+				<TodoInput
+					date={date}
+					todo={todo}
+					onChangeDate={changeDate}
+					onChangeTodo={changeTodo}
+					onSubmit={submitTodo}
+				/>
+				<Space />
+				<TodoList data={todos} onRemove={removeTodo} />
+			</Contents>
+		</Wrapper>
+	);
```

Comment:
```suggestion
	return (
		<Wrapper>
			<Title>리액트-투두</Title>
			<Contents>
				<TodoInput
					onSubmit={handleAdd}
				/>
				<Space />
				<TodoList {...{todos}} onDelete={handleDelete} />
			</Contents>
		</Wrapper>
	);
```

1. 바뀐 컴포넌트 네임은 아래 styled-components 코드에 달린 리뷰를 확인해주세요.
2. onSubmit 네이밍 좋네요 👍 props로 핸들러를 넘길 때 props 이름에는 접두사 ```on```을 붙입니다.
핸들러를 받아서 쓸 ```TodoItem``` 입장에서 위 핸들러로 form을 제출하기 때문에 네이밍 ```onSubmit```으로 해주신 거 너무 좋네요~
3. 굳이 todos를 data라는 이름의 props로 넘길 필요는 없을 것 같아요~
props 이름과 넘기려는 값을 담은 변수의 이름이 같다면 spread operator를 이용해 위와 같이 넘겨주시면 좋아요.